cmd/ci-operator-configresolver/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+	"github.com/openshift/ci-tools/pkg/load"
+	"github.com/openshift/ci-tools/pkg/registry"
+	types "github.com/openshift/ci-tools/pkg/steps/types"
+)
+
+type options struct {
+	resolver   registry.Resolver
+	configPath string
+	regPath    string
+}
+
+func resolveConfig(o *options) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusNotImplemented)
+			w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+		org := r.URL.Query().Get("org")
+		if org == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("org query missing or incorrect"))
+			return
+		}
+		repo := r.URL.Query().Get("repo")
+		if repo == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("repo query missing or incorrect"))
+			return
+		}
+		branch := r.URL.Query().Get("branch")
+		if branch == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("branch query missing or incorrect"))
+			return
+		}
+		variant := r.URL.Query().Get("variant")
+		info := config.Info{
+			Org:     org,
+			Repo:    repo,
+			Branch:  branch,
+			Variant: variant,
+		}
+		info.Filename = filepath.Join(o.configPath, branch, info.Basename())
```

Comment:
Is this directory structure possible with the ConfigMap structure we have?

https://github.com/openshift/release/blob/725610d085583d0af88eafbe292cbb56897376b9/core-services/prow/02_config/_plugins.yaml#L1198-L1259

Reply:
For our configmaps that contain the ci-operator configs, it should work fine. The names of the keys are filenames and when mounted everything _should_ work as expected. I can try this out on a test cluster to make sure though.

Reply:
I don't think we can create separate directories for each branch. Some of the patterns in the configuration have wildcards for the branch part, or for the entire file name.